apps/server/src/modules/authentication/uc/account.uc.spec.ts

```diff
@@ -18,53 +45,188 @@ describe('AccountUc', () => {
+		});
+		mockAdminUser = userFactory.buildWithId({
+			school: mockSchool,
+			roles: [new Role({ name: 'admin', permissions: ['TEACHER_EDIT', 'STUDENT_EDIT'] })],
+		});
+		mockTeacherUser = userFactory.buildWithId({
+			school: mockSchool,
+			roles: [new Role({ name: 'teacher', permissions: ['STUDENT_EDIT'] })],
+		});
+		mockDemoTeacherUser = userFactory.buildWithId({
+			school: mockSchool,
+			roles: [new Role({ name: 'demoTeacher', permissions: ['STUDENT_EDIT'] })],
+		});
+		mockOtherTeacherUser = userFactory.buildWithId({
+			school: mockSchool,
+			roles: [new Role({ name: 'teacher', permissions: ['STUDENT_EDIT'] })],
+		});
+		mockStudentUser = userFactory.buildWithId({
+			school: mockSchool,
+			roles: [new Role({ name: 'student', permissions: [] })],
+		});
+		mockDifferentSchoolAdminUser = userFactory.buildWithId({
+			school: mockotherSchool,
+			roles: [new Role({ name: 'admin', permissions: ['TEACHER_EDIT', 'STUDENT_EDIT'] })],
 		});
-	});
 
-	describe('create', () => {
-		it('should return an account', async () => {
-			const account = await uc.create(mockAccount);
-			expect(account).toEqual(mockAccount);
+		mockSuperheroAccount = accountFactory.buildWithId({ user: mockSuperheroUser, password: 'hjkl' });
+		mockTeacherAccount = accountFactory.buildWithId({ user: mockTeacherUser, password: 'asdf' });
+		mockDemoTeacherAccount = accountFactory.buildWithId({ user: mockDemoTeacherUser, password: 'vbnm' });
+		mockOtherTeacherAccount = accountFactory.buildWithId({ user: mockOtherTeacherUser, password: 'uiop' });
+		mockAdminAccount = accountFactory.buildWithId({ user: mockAdminUser, password: 'qwer' });
+		mockStudentAccount = accountFactory.buildWithId({ user: mockStudentUser, password: '1234' });
+		mockDifferentSchoolAdminAccount = accountFactory.buildWithId({
+			user: mockDifferentSchoolAdminUser,
+			password: 'yxcv',
 		});
 	});
 
-	describe('update', () => {
-		it('should return an account', async () => {
-			const account = await uc.update(mockAccount);
-			expect(account).toEqual(mockAccount);
+	describe('changePasswordForUser', () => {
+		it('should throw if account does not exist', async () => {
+			await expect(accountUc.changePasswordForUser('currentUser', 'targetUser', 'DummyPasswd!1')).rejects.toThrow(
+				EntityNotFoundError
+			);
+		});
+
+		it('should throw if user does not exist', async () => {
+			await expect(accountUc.changePasswordForUser('currentUser', 'targetUser', 'DummyPasswd!1')).rejects.toThrow(
+				EntityNotFoundError
+			);
+		});
+
+		it('should throw if password is weak', async () => {
+			await expect(
+				accountUc.changePasswordForUser(mockAdminAccount.user.id, mockTeacherAccount.user.id, 'weak')
+			).rejects.toThrow();
+		});
+
+		it('should accept strong password, update it and force renewal', async () => {
+			const previousPasswordHash = mockTeacherAccount.password;
+			expect(mockTeacherAccount.user.forcePasswordChange).toBeFalsy();
```

Comment:
Please don't use `toBeFalsy()` unless you really don't care about a boolean type. If it is a boolean use something like `toBe(false)` or `toEqual(false)`.

Reply:
It was actually `undefined` and not `false`, but I replaced it anyways.